tests/integration/test_data/bundler_everything_present/container/Containerfile

```diff
@@ -0,0 +1,25 @@
+FROM docker.io/ruby:3.3
+
+# Test disabled network access
+RUN if curl -IsS www.google.com; then echo "Has network access!"; exit 1; fi
+
+# Print cachi2 env vars file
+RUN cat /tmp/cachi2.env
+
+# Check bundler deps
+RUN ls /tmp/bundler_everything_present-output/deps/bundler
+
+# Check content of source repository folder
+RUN ls /tmp/bundler_everything_present-source/
+
+# This should be a COPY, but the source code and Containerfile are in different directories
+RUN cp -r /tmp/bundler_everything_present-source /src
+
+WORKDIR /src
+# Bundler would try and install whichever version was used to generate Gemfile.lock.
+# This will cause bundler to attempt to download an earlier version even if
+# just microversions diverged. This, in turn, would cause a build to fail.
+# Running 'bundle _<version>_ install' is supposed to attempt to run installation
+# with this specific version. The extra code below ensures that any present
+# version is used:
```

Comment:
Interesting behaviour....which means that it's not just the problem of our e2e test, users will suffer from this as well in hermetic envs and we surely don't want Bundler to fail the installation due to a no network access! I don't think what follows is the correct fix, instead I think we should override the `BUNDLE_VERSION` [1] variable along with the other ones and document.

[1] https://bundler.io/v2.5/man/bundle-config.1.html#LIST-OF-AVAILABLE-KEYS
[2] https://github.com/rubygems/rubygems/issues/6613
[3] https://github.com/rubygems/rubygems/pull/6824/

Reply:
> Can we do this? We can detect a version that is used in a fetch environment, but there is no guarantee that this same version is used in build environment. 

Yes we can! The reason I suggested is the keyword `system` that the variable accepts. From the docs:

> system will use the system version of Bundler

Reply:
Missed the 'system' part yesterday, it does work after all.